features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/voicemessages/timeline/VoiceMessagePresenter.kt

```diff
@@ -102,7 +105,19 @@ class VoiceMessagePresenter @AssistedInject constructor(
                     if (playerState.isPlaying) {
                         player.pause()
                     } else {
-                        scope.launch { play.runUpdatingState { player.play() } }
+                        scope.launch {
+                            play.runUpdatingState(
+                                errorTransform = {
+                                    it.apply {
```

Comment:
The `apply` is not necessary, I would simply write

```kotlin
errorTransform = {
    analyticsService.trackError(
        VoiceMessageException.PlayMessageError("Error while trying to play voice message", it)
    )
    it
},
```

to improve readability.

Reply:
I had it exactly like that before but it looked odd to my eyes so I switched to using `apply` 😆